### ISSUE TITLE
Regex tweak for GB VAT nos.

### ISSUE THREAD

**Issue by lnpbk** [ddeboer/vatin#23]
Forgive me if I'm wrong, but should the regex pattern for GB VAT ids be wrapped in parens?

`'GB' => '(\d{9}|\d{12}|(GD|HA)\d{3})'`

Currently if you supply a valid VAT number with any random string after it, it still passes validation. Wrapping the pattern in parentheses fixes the issue, but wanted to raise this here first in case I was missing something obvious.

**Comment by willemstuursma** [comment]
A `^` and `$` are added to the regexp in https://github.com/ddeboer/vatin/blob/master/src/Validator.php#L96.

**Comment by lnpbk** [comment]
@willemstuursma Maybe I'm missing something then, as I've just tested this:

```
$validator = new Validator;
var_dump($validator->isValid('GB999999973'));               // true
var_dump($validator->isValid('GB999999973dsflksdjflsk'));   // true
```

**Comment by sstok** [comment]
Confirmed the whole pattern needs to be wrapped in a group otherwise the `$` is matched for only the last part. Regexes are tricky :) as a general rule of thump if everything must match begin to end wrap the whole regex in a group.

**Comment by lnpbk** [comment]
@sstok Yeah, I'm no expert with regex, hence wanting to raise discussion

**Comment by willemstuursma** [comment]
Sorry, I was a bit to quick then!

**Comment by willemstuursma** [comment]
Should add some `(?:` at the start and `)` at the end, and some unit tests of course.

**Comment by ddeboer** [comment]
Wowsers, good catch @lnpbk!

**Comment by ddeboer** [comment]
Fixed in #24.
